hmpps-prisoner-search/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/search/services/attributesearch/AttributeSearchIntegrationTest.kt

```diff
@@ -0,0 +1,146 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.search.services.attributesearch
+
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.springframework.test.web.reactive.server.WebTestClient
+import uk.gov.justice.digital.hmpps.prisonersearch.common.model.SyncIndex
+import uk.gov.justice.digital.hmpps.prisonersearch.search.AbstractSearchDataIntegrationTest
+import uk.gov.justice.digital.hmpps.prisonersearch.search.model.PrisonerBuilder
+import uk.gov.justice.digital.hmpps.prisonersearch.search.repository.PrisonerRepository
+
+class AttributeSearchIntegrationTest : AbstractSearchDataIntegrationTest() {
+
+  private val prisonerBuilders = mutableListOf<PrisonerBuilder>()
+
+  // Although we want the super class to create a new index before the tests run, we don't want to load any data into it, so we can manage prisoners on a test by test basis
+  override fun loadPrisonerData() {}
+
+  private fun PrisonerRepository.delete(prisonerNumber: String) = delete(prisonerNumber, SyncIndex.GREEN)
+
+  private fun WebTestClient.attributeSearch(body: String) =
+    post()
+      .uri("/attribute-search")
+      .headers(setAuthorisation(roles = listOf("ROLE_PRISONER_SEARCH")))
+      .header("Content-Type", "application/json")
+      .bodyValue(body)
+      .exchange()
+      .expectStatus().isOk
+
+  private fun loadPrisoners(vararg prisoners: PrisonerBuilder) =
+    prisonerBuilders.apply {
+      addAll(prisoners)
+      loadPrisonersFromBuilders(this)
+    }
+
+  @BeforeEach
+  fun `clear search repository`() {
+    prisonerBuilders.onEach { prisonerRepository.delete(it.prisonerNumber) }.clear()
+    waitForPrisonerLoading(0)
+  }
+
+  @Test
+  fun `single string matcher`() {
+    loadPrisoners(
+      PrisonerBuilder(prisonerNumber = "A1234AA", firstName = "John"),
+      PrisonerBuilder(prisonerNumber = "B1234BB", firstName = "Jeff"),
+    )
+
+    webTestClient.attributeSearch(
+      """
+        {
+          "queries": [
+            {
+              "matchers": [ 
+                {
+                  "type": "String",
+                  "attribute": "firstName",
+                  "condition": "IS",
+                  "searchTerm": "John"
+                }
+              ]
+            }
+          ]
+        }
+      """.trimIndent(),
+    )
+      .expectBody()
+      .jsonPath("$.content.length()").isEqualTo(1)
```

Comment:
not a great fan of this approach of comparing - if there is not one row then not sure you'll get a good comparison out of what is different and why.  I've extracted all the prisoner numbers out before and then compared the list - see https://github.com/ministryofjustice/hmpps-prisoner-search/blob/9cb56734c3dae1628432e20644a055530c69f55d/hmpps-prisoner-search-indexer/src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/indexer/resource/PrisonerDifferencesResourceIntTest.kt#L56

Reply:
I've actually switched to that method in my next PR